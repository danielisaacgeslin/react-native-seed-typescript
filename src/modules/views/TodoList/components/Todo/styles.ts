import { StyleSheet } from 'react-native';
import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  container: {
    marginTop: 5,
    marginBottom: 5
  },
  title: {
    fontSize: 19,
    color: STYLE.COLOR.PRIMARY
  },
  description: {
    fontSize: 15
  },
  date: {
    fontSize: 13
  }
});
