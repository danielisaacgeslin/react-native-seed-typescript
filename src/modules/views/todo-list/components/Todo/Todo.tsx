import React, { memo } from 'react';
import { View, Text } from 'react-native';

import { TodoModel } from '../../../../models';
import { ErrorBoundary } from '../../../shared/components';
import styles from './styles';

export interface ITodoProps {
  todo: TodoModel.ITodo;
}

const Todo = ({ todo }: ITodoProps) => {
  return (
    <ErrorBoundary>
      <View style={styles.container}>
        <Text style={styles.title}>{todo.title.toUpperCase()}</Text>
        <Text style={styles.description}>{todo.description}</Text>
        <Text style={styles.date}>created at: {String(todo.createdAt)}</Text>
      </View>
    </ErrorBoundary>
  );
};

export default memo(Todo);
