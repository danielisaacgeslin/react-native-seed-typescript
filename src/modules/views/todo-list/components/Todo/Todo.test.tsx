import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import Todo, { ITodoProps } from './Todo';
import { getTodo_1 } from '../../../../../test/entities';

describe('Login', () => {
  let props: ITodoProps;
  let wrapper;

  global.console.error = () => null;
  beforeEach(() => {
    props = { todo: getTodo_1() };
    wrapper = mount(<Todo {...props} />);
  });

  it('should render', () => {
    expect(create(wrapper).toJSON()).toMatchSnapshot();
  });
});
