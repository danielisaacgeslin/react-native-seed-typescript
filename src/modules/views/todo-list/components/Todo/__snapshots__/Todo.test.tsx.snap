// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render 1`] = `
<View
  style={
    Object {
      "marginBottom": 5,
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1745D5",
        "fontSize": 19,
      }
    }
  >
    I AM A TODO
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    i am a todo description
  </Text>
  <Text
    style={
      Object {
        "fontSize": 13,
      }
    }
  >
    created at: 
    2018-05-11
  </Text>
</View>
`;
